app/use_cases/create_voting_report.rb

```diff
@@ -0,0 +1,50 @@
+class CreateVotingReport
+
+  attr_reader :batches, :votes, :report
+
+  def initialize(batches, votes)
+    @batches = batches
+    @votes = votes
+    @report = {}
+  end
+
+  def perform
+    batches.sort_by! &:start_date
```

Comment:
You could get the database to sort the batches when you retrieve them `Batch.where(..).order(starts_at: :asc)` 

It is a performance improvement - and in this case it probably is fine as the datasets are quite small, but in general good practice to do filtering and sorting in the database before the objects need to be loaded into ruby, as ruby is many times slower and hungry on memory than the database.